_posts/2019-05-10-server-rendering-responsively.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: epic
+title: "Server-Rendering Responsively"
+date: "2019-04-22"
+author: [steve-hicks]
+categories: [react, html, web]
+---
+
+We use server-side rendering (SSR) to deliver every page you hit on [artsy.net](https://artsy.net). We decided on
+using SSR for many reasons, amongst them performance. (TODO: Maybe more about why we chose SSR, but I don't know
+much about that history.)
+
+We've also built our site using responsive design, so you get a browsing experience optimized for your device.
+
+Combining SSR and responsive design is a non-trivial problem. There are many concerns to manage, and they are
+sometimes in conflict with each other. We server render for performance reasons, but we also want to be sure our
+app is performant when your browser takes over, all while optimizing accessibility and SEO.
+
+This article describes the tools we use on [artsy.net](https://artsy.net) to combine SSR and responsive design.
+
+<!-- more -->
+
+## Tool 1: [`styled-system`](https://styled-system.com)
+
+We handle the majority of responsive styling differences with
+[`styled-system`](https://styled-system.com/responsive-styles). This has been a really great addition to our
+toolbox. Here's a component that would render a `div` (`Box`) with a width of 50% at our mobile breakpoint, 75% at
+our tablet breakpoint, and 100% for anything larger:
+
+```xml
+<Box width={["50%", "75%", "100%"]}>
+  ...
+</Box>
+```
+
+As developers, we love this experience. We can make subtle differences to components across breakpoints with very
+little code and effort.
+
+We're working on another blog post that digs deeper into how we use `styled-system` within
+[our design system](https://palette.artsy.net/). You can also
+[poke around our source](https://github.com/artsy/reaction/blob/f3dabb884d616c5c42bbd47b1432dc2a9456b8ca/src/Apps/Search/Components/SearchResultsSkeleton/Header.tsx#L6)
+to see how much we've embraced `styled-system`'s responsive styles. (In that example, `pr` means `padding-right`
+and `pl` means `padding-left`.)
+
+There's one type of challenge with building a responsive app that `styled-system` can't solve: when we need to emit
+different layouts across different breakpoints. In this case, we need something that can render very different
+component sub-trees. We couldn't find an approach that satisfied our needs, so we wrote our own.
+
+## Tool 2: [`@artsy/react-responsive-media`](https://github.com/artsy/react-responsive-media)
+
+First off, an announcement: we've just released `react-responsive-media` version 1.0! (TODO: release v1.0 😬 or
+change this announcement)
```

Comment:
This was brought up yesterday, prompted by @joeyAghion's post on naming. Thinking, maybe we could come up with a better name before publishing this post? @alloy and I have briefly chatted about it in comments. 

(Related, I'd like to update the examples structure a bit inside of the lib to make it a bit less "kitchen sink" like, and separate things into a "hello world" and "ssr" folder. RRM can just as easily be used on the client as any other responsive-media lib, with the added bonus that RRM has a better API.)

Reply:
I totally defer to you all about that. Once I get this thing in a shippable state, I'll push for a decision.